addons/mail/models/mail_tracking_value.py

```diff
@@ -64,7 +64,7 @@ def create_tracking_values(self, initial_value, new_value, col_name, col_info, t
             })
         elif col_info['type'] == 'selection':
             values.update({
-                'old_value_char': initial_value and dict(col_info['selection'])[initial_value] or '',
+                'old_value_char': initial_value and dict(col_info['selection']).get(initial_value, initial_value) or '',
```

Comment:
I know nothing about this problem, but it feels strange to use the field name as default value rather than empty string?

Reply:
Ok I think I understand better, so the goal is to have the selection "key" rather than the selection "string" as a best effort value if the selection key somehow doesn't exist?

I'm not sure if it will be very clear for the user, but I assume there is no way to correctly know both the old and new selection keys in the problematic scenario?

Reply:
It can also be the case outside of the upgrade. If for any reason, the current value of the selection field is unknown by the ORM, the user is stuck and can't change the value (using the UI).